src/js/core/UpdateStream.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable } from "rxjs/Observable";
+import compareVersions from "compare-versions";
+import { request, RequestResponse } from "@dcos/http-service";
+import { BehaviorSubject } from "rxjs";
+
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+export const UpdateStreamType: symbol = Symbol("UpdateStreamType");
+const CHECK_DELAY: number = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION_SETTING: string = "dcosUIVersion";
+const DISMISSED_VERSION: string = "dismissedVersion";
+const LAST_TIME_CHECK: string = "lastTimeCheck";
+
+interface ContentTypeObject {
+  action: string;
+  actionType: string;
+  entity: string;
+  version: string;
+}
+
+interface DCOSUserSettings {
+  dcosUIVersion: DCOSUIVersion;
+  [key: string]: DCOSUIVersion;
+}
+
+interface DCOSUIVersion {
+  dismissedVersion: string;
+  lastTimeCheck: number;
+  [key: string]: string | number;
+}
+
+export function showNotification(newVersion: string): void {
+  // TODO: Move inside UpdateServiceBanner
+  alert(newVersion);
+}
+
+function getVersionFromVersionObject(versionObject: any): string {
+  return Object.keys(versionObject.response.results)[0];
+}
+
+function getFromLocalStorage(key: string): any {
```

Comment:
Do we really have no idea what is returned here?

Reply:
We do, but if I write `string | number` for example, then the `localStorageDismissedVersion` definition complains that
```
[ts]
Type 'BehaviorSubject<string | number>' is not assignable to type 'BehaviorSubject<string>'.
  Types of property 'value' are incompatible.
    Type 'string | number' is not assignable to type 'string'.
      Type 'number' is not assignable to type 'string'. [2322]
```
.

Reply:
Because the BehaviorSubject where you use this function only expects strings and no numbers. Either have this function return string or have the BehaviourSubject handle string | number.

In general, using the fewest amount, but the most specific types in a union type is a good rule of thumb

Reply:
I don't like the solution of making both `BehaviorSubject`s `string | number`, but I don't see any other way, so I guess it's okay.